lib/puppet/functions/libkv/delete.rb

```diff
@@ -1,86 +1,117 @@
+  #
+  #   * `backend`: String.  Name of the backend to use.
+  #
+  #     * When present, must match a key in the `backends` option of the
+  #       merged options Hash.
+  #     * When absent and not specified in `libkv::options`, this function
+  #       will look for a 'default.xxx' backend whose name matches the
+  #       `resource` option.  This is typically the catalog resource id of the
+  #       calling Class, specific defined type instance, or defined type.
+  #       If no match is found, it will use the 'default' backend.
+  #
+  #   * `environment`: String.  Puppet environment to prepend to keys.
+  #
+  #     * When set to a non-empty string, it is prepended to the key used in
+  #       the backend operation.
+  #     * Should only be set to an empty string when the key being accessed is
+  #       truly global.
+  #     * Defaults to the Puppet environment for the node.
+  #
+  #   * `resource`: String.  Name of the Puppet resource initiating this libkv
+  #     operation
+  #
+  #     * Required when `backend` is not specified and you want to be able
+  #       to use more than the `default` backend.
+  #     * String should be resource as it would appear in the catalog or
+  #       some application grouping id
+  #
+  #       * 'Class[<class>]' for a class, e.g.  'Class[Mymodule::Myclass]'
+  #       * '<Defined type>[<instance>]' for a defined type instance, e.g.,
+  #         'Mymodule::Mydefine[myinstance]'
+  #
+  #     * Catalog resource id cannot be reliably determined automatically.
+  #       Appropriate scope is not necessarily available when a libkv function
+  #       is called within any other function.  This is problematic for heavily
+  #       used Puppet built-in functions such as `each`.
+  #
+  #  * `softfail`: Boolean. Whether to ignore libkv operation failures.
   #
-  # @return [Any] The value in the underlying backing store
+  #    * When `true`, this function will return a result even when the operation
+  #      failed at the backend.
+  #    * When `false`, this function will fail when the backend operation failed.
+  #    * Defaults to `false`.
   #
+  # @raise [ArgumentError] If the key or merged backend config is invalid
+  #
+  # @raise [LoadError] If the libkv adapter cannot be loaded
+  #
+  # @raise [RuntimeError] If the backend operation fails, unless 'softfail' is
+  #   `true` in the merged backend options.
+  #
+  # @return [Boolean] `true` when backend operation succeeds; `false` when the
+  #   backend operation fails and 'softfail' is `true` in the merged backend
+  #   options
   #
   dispatch :delete do
-    param 'Hash', :parameters
+    required_param 'String[1]', :key
+    optional_param 'Hash',      :options
   end
 
+  def delete(key, options={})
+    # key validation difficult to do via a type alias, so validate via function
+    call_function('libkv::validate_key', key)
 
+    # add libkv 'extension' to the catalog instance as needed
+    call_function('libkv::add_libkv')
```

Comment:
Yeah, it seems like this should be `libkv::load` (or something like that) now that I see it in action.